### PR TITLE
[CALCITE-6623] The MongoDB adapter throws a java.lang.ClassCastException when Decimal128 or Binary types are used, or when a primitive value is cast to a string

### PR DESCRIPTION
**Pull request by dssysolyatin**

### CODE REVIEW

File: mongodb/src/main/java/org/apache/calcite/adapter/mongodb/MongoEnumerator.java [hunk 1]
```diff
@@ -119,8 +127,34 @@ static Function1<Document, Object> getter(
...
+   *
+   * <h3>Conversion Details</h3>:
+   *
+   * <p>If the target type is one of the following, the method performs specific conversions:
+   * <ul>
+   *   <li>`Long`: Converts a `Date` or `BsonTimestamp` object into the respective epoch time
+   *   (milliseconds).
+   *   <li>`BigDecimal`: Converts a `Decimal128` object into a `BigDecimal` instance.
+   *   <li>`String`: Converts arrays to string and uses `String.valueOf(o)` for other objects.
+   *   <li>`ByteString`: Converts a `Binary` object into a `ByteString` instance.
```
[the remark this answers is not in the document]
Comment (dssysolyatin) [reply mongodb/src/main/java/org/apache/calcite/adapter/mongodb/MongoEnumerator.java:152]: @mihaibudiu 
This code is not related to RexLiteral; it is related to plan execution. When Calcite receives objects from the MongoDB client, it should convert them to calcite runtime object.

I also added a section to the documentation about converting primitive types and boxed primitives.

File: mongodb/src/test/java/org/apache/calcite/adapter/mongodb/MongoAdapterTest.java [hunk 2]
```diff
@@ -778,6 +738,50 @@ private void checkPredicate(int expected, String q) {
...
+    assertModel(MODEL)
+        .query("select cast(_MAP['binaryData'] AS binary) from \"mongo_raw\".\"datatypes\"")
+        .returnsCount(1)
+        .returns(resultSet -> {
+          try {
+            resultSet.next();
+            //CHECKSTYLE: IGNORE 1
+            assertThat(new String(resultSet.getBytes(1), StandardCharsets.UTF_8), is("binaryData"));
+          } catch (Throwable e) {
+            throw new RuntimeException(e);
```
[the remark this answers is not in the document]
Comment (dssysolyatin) [reply mongodb/src/test/java/org/apache/calcite/adapter/mongodb/MongoAdapterTest.java]: input.next() can throw a SQLException, which is a checked exception, so it needs to be handled. I also modified the code slightly to use a test utility class, like this:
```
catch (SQLException e) {
            throw TestUtil.rethrow(e);
          }
```

**Comment by dssysolyatin** [comment]
@mihaibudiu, if the recent changes are okay with you, I will merge the PR

**Comment by mihaibudiu** [comment]
I have already approved.

**Pull request merged**